pypeit/specobj.py

```diff
@@ -432,23 +442,26 @@ def apply_flux_calib(self, sens_dict, exptime, telluric_correct=False, extinct_c
             if attr+'_WAVE' not in self._data.keys():
                 continue
             msgs.info("Fluxing {:s} extraction for:".format(attr) + msgs.newline() + "{}".format(self))
-            #
-            #try:
-            #    wave = np.copy(np.array(extract['WAVE_GRID']))
-            #except KeyError:
+
             wave = self[attr+'_WAVE']
-            wave_sens = sens_dict['wave']
-            sensfunc = sens_dict['sensfunc'].copy()
+            # Interpolate the sensitivity function onto the wavelength grid of the data
 
-            # Did the user request a telluric correction from the same file?
-            if telluric_correct and 'telluric' in sens_dict.keys():
+            # TODO Telluric corrections via this method are deprecated
+            # Did the user request a telluric correction?
+            if telluric is not None:
                 # This assumes there is a separate telluric key in this dict.
-                telluric = sens_dict['telluric']
                 msgs.info('Applying telluric correction')
                 sensfunc = sensfunc * (telluric > 1e-10) / (telluric + (telluric < 1e-10))
 
-            sensfunc_obs = interpolate.interp1d(wave_sens, sensfunc, bounds_error=False,
-                                                      fill_value='extrapolate')(wave)
+            sensfunc_obs = np.zeros_like(wave)
+            wave_mask = wave > 1.0  # filter out masked regions or bad wavelengths
+            try:
+                sensfunc_obs[wave_mask] = interpolate.interp1d(wave_sens, sensfunc, bounds_error=True)(wave[wave_mask])
+            except ValueError:
+                msgs.error("Your data extends beyond the bounds of your sensfunc. " + msgs.newline() +
+                           "Adjust the par['sensfunc']['extrap_blu'] and/or par['sensfunc']['extrap_red'] to extrapolate "
+                           "further and recreate your sensfunc.")
```

Comment:
When I'm debugging, it's much easier if the code faults where the problem is than to fault in an except block.  I generally try to reserve try/except blocks for cases when I know exactly why the line of code faulted and how to handle the fault.  So doing this may cause things to be difficult to debug, but maybe not.  I'd have to experience the fault first to get a sense of this.

Reply:
I need to print out this error message if a value is returned so the user can know why it faulted. So I think I'm making it the easiest to debug. If you can suggest a better implementation I'm happy to use it, but this seems exactly what you want try except for, i.e. if this specific extrapolation exception occurs print out this message.

Reply:
I'd have to encounter the problem before really being able to make a judgement call.